addons/hw_drivers/iot_handlers/interfaces/DisplayInterface_L.py

```diff
@@ -19,31 +16,51 @@ class DisplayInterface(Interface):
 
     def get_devices(self):
         display_devices = {}
-        x_screen = 0
-        hdmi_port = {'hdmi_0' : 2}
-        rpi_type = GPIO.RPI_INFO.get('TYPE')
-        # RPI 3B+ response on for booth hdmi port
-        if 'Pi 4' in rpi_type:
-            hdmi_port.update({'hdmi_1': 7})
 
         try:
-            for hdmi in hdmi_port:
-                power_state_hdmi = Vcgencmd().display_power_state(hdmi_port.get(hdmi))
-                if power_state_hdmi == 'on':
-                    iot_device = {
-                        'identifier': hdmi,
-                        'name': 'Display hdmi ' + str(x_screen),
-                        'x_screen': str(x_screen),
-                    }
-                    display_devices[hdmi] = iot_device
-                    x_screen += 1
-        except subprocess.CalledProcessError:
-            _logger.warning('Vcgencmd "display_power_state" method call failed')
+            # Get xrandr output and parse lines
+            # 'check_output' method returns a bytes object, so we need to decode it to a string and split it by lines
+            # 'split' method is used to get only the last element of each line : 'number of monitors' then 'identifiers'
+            os.environ['DISPLAY'] = ":0"
+            parsed_xrandr_output = [
+                o.split()[-1]
+                for o in subprocess.run(['xrandr', '--listactivemonitors'], check=True, stdout=subprocess.PIPE)
```

Comment:
To avoid a subprocess call we can use another module: `screeninfo`
https://pypi.org/project/screeninfo/

>>> import screeninfo
>>> screeninfo.get_monitors()
[Monitor(x=0, y=0, width=1920, height=1080, width_mm=480, height_mm=270, name='HDMI-1', is_primary=True)]